### DISCUSSION TITLE
Can I use Nixtla's transformer implementations for multivariate time series?

### DISCUSSION THREAD

**Discussion by gabriead** [Nixtla/neuralforecast#683] [Q&A]
Hi community,
I would like to use Nixtla for my Master thesis. I would like to compare a bunch of transformer based models on my multivariate data. From the documentation it is unclear to me if I can use every of the transformer models for multi-variate time series prediciton as well?

**Comment by kdgutier** [comment]
Hey @gabriead, 

By multivariate time series
- Do you refer to forecasting with exogenous variables?
- Do you refer to Vector Autorregresive approach?

If the answer is the first for the moment the only available Transformer for exogenous is TFT.
https://nixtla.github.io/neuralforecast/examples/forecasting_tft.html

**Reply by gabriead** [reply]
Hey @kdgutier,
thank's for the super quick reply. Just to clarify I want to forecast a physiological parameter based on other physiological parameters so from my understanding those features would be exogenous variables to the target, correct? I haven't heard of the Vector Autoregressive approach yet, could you maybe elaborate a bit on that in regards to my use case?
So for the first case I could use TFT ( how does the Nixtla implementation differ from the Pytorch TFT?) from Nixtla and would have to use other Github Repos (Autoformer, Informer, Scaleformer, ETSFormer ...) and adapt them myself correct?

**Reply by ams015** [reply]
Hi @kdgutier,
So would the rest of the transformers use a VAR approach?

**Comment by kdgutier** [comment]
Here is the library's [contents index](https://nixtla.github.io/neuralforecast/examples/models_intro.html):

These models in NeuralForecast library support exogenous variables:
NBEATSx, TFT, MLP, NHITS, LSTM, GRU, RNN, DeepAR.

Transformer-based models (except for TFT) do not use exogenous variables, the methods are:
Informer, Autoformer, VanillaTransformer.

**Reply by gabriead** [reply]
So by Informer, Autoformer, VanillaTransformer models you mean specifically the implementations in Nixtla do not provide exogenous variables?
